### ISSUE TITLE
youtube - Download failed: HTTP is not supported

### ISSUE THREAD

**Issue by auouymous** [gpodder/gpodder#625]
It looks like http_request() always uses http even for https URLs, this fixes that.

```
diff --git a/src/gpodder/util.py b/src/gpodder/util.py
index 7103bd7a..abf32bac 100644
--- a/src/gpodder/util.py
+++ b/src/gpodder/util.py
@@ -1402,7 +1402,10 @@ def format_seconds_to_hour_min_sec(seconds):
 
 def http_request(url, method='HEAD'):
     (scheme, netloc, path, parms, qry, fragid) = urllib.parse.urlparse(url)
-    conn = http.client.HTTPConnection(netloc)
+    if scheme == 'https':
+        conn = http.client.HTTPSConnection(netloc)
+    else
+        conn = http.client.HTTPConnection(netloc)
     start = len(scheme) + len('://') + len(netloc)
     conn.request(method, url[start:])
     return conn.getresponse()

```

And this enables https for youtube.

```
diff --git a/src/gpodder/youtube.py b/src/gpodder/youtube.py
index c3e59320..2c87647a 100644
--- a/src/gpodder/youtube.py
+++ b/src/gpodder/youtube.py
@@ -116,7 +116,7 @@ def get_real_download_url(url, preferred_fmt_ids=None):
     vid = get_youtube_id(url)
     if vid is not None:
         page = None
-        url = 'http://www.youtube.com/get_video_info?&el=detailpage&video_id=' + vid
+        url = 'https://www.youtube.com/get_video_info?&el=detailpage&video_id=' + vid
 
         while page is None:
             req = util.http_request(url, method='GET')

```

**Comment by auouymous** [comment]
I posted the wrong patch, first one needs a colon after the 'else'.

**Comment by pieska** [comment]
Thanks for the fix

**Comment by RoyalT17** [comment]
I am a total Noob, how do I apply the fix to make it work with YouTube again?

**Comment by pieska** [comment]
find the two files youtube.py and util.py on your system (on my linux system they are in /usr/lib/python3.7/site-packages/gpodder), take a text editor and edit the files by replacing the lines marked with "-" with the "+" ones without the leading "+". you should back up the files before changing them..

**Comment by auouymous** [comment]
if you have `patch`, download https://github.com/gpodder/gpodder/pull/626.patch and place it in /root/, then enter the directory containing those two files and type `patch -p3 < /root/626.patch`

**Comment by RoyalT17** [comment]
guess I should have been more clear, I have gPodder on Windows. Thanks for the Advise. I might be able to  work my way through it. 👍

**Comment by RoyalT17** [comment]
Thanks to auouymous and pieska, I figured it out! awesome work and directions! Much Kudos

**Comment by elelay** [comment]
Thanks for the fix!
Seems like YouTube plugged the plug on http access.
I'll try to make a release soon.

**Comment by Jennifer202** [comment]
Oh I have tears of joy. Thank you. If a person finds this that looks at the above fix and are not a coder (like me) but finds it a bit much for them I will put down very simple directions here for how to fix it. I have not used windows in a long time sorry for no help there.

1) You have to find where 2 scripts (text files) are one is "youtube.py" and the other is "util.py" they may be locked down so you will me to be "root" or "admin" to edit them  I have a Ubuntu 18.04 and the file for me was at /usr/lib/python3/dist-packages/gpodder/ the above person found it at /usr/lib/python3.7/site-packages/gpodder. You may need to do a search for it but it should be in one of the "python..." folders there. 

2) Fix "util.py", first make a copy of it, just in case something goes wrong
Open it with a text editor (anything that opens text files like leaf pad will work) do a search for "conn = http.client.HTTPConnection(netloc)" then, in its place, on 4 different lines put:
    if scheme == 'https':
        conn = http.client.HTTPSConnection(netloc)
    else:
        conn = http.client.HTTPConnection(netloc)
The spaces in front are not needed per-say, they are there to make it readable to humans. 

3) Fix "youtube.py", make a copy
With your text editor search for "url = 'http://www.youtube.com/get_video_info?&el=detailpage&video_id=' + vid" 
See the the first part "url = 'http:// ......" just stick a little 's' there like this "url = 'https://...."
or past in:
url = 'https://www.youtube.com/get_video_info?&el=detailpage&video_id=' + vid

Move any "backup" copies of the script that you made out of the the folder (if you kept them in there) and give it a test run. If all is good you can delete your backup scripts.

Hope this helps someone if it was not for auouymous and pieska above I would not have been able to fix it myself.

For the spiders: "HTTP is not supported" "gpodder"
Sadly when I did a search with duck,duck go then with Bing (who now owns this site) and even google proper; this page did not show up. With the other post https://github.com/gpodder/gpodder/issues/628 (that google found) linking to this one I hope more people find this.

**Comment by jskrdla** [comment]
I'm on Windows, and I tried the suggestion above (updating util.py and youtube.py). I'm probably missing a step, but now when I try to run gPodder nothing happens. What am I missing? I even moved my backup files back in and still get nothing.

**Comment by mads5408** [comment]
> I posted the wrong patch, first one needs a colon after the 'else'.

Remember this jskrdia

**Comment by jskrdla** [comment]
I put in the colon. Perhaps I'm missing something else...

**Comment by BeatLink** [comment]
Heres a oneliner code for those using flatpak (download the patch from https://patch-diff.githubusercontent.com/raw/gpodder/gpodder/pull/626.patch to your downloads folder: 

`cd /var/lib/flatpak/app/org.gpodder.gpodder/x86_64/stable/active/files/lib/python3.5/site-packages/gpodder && sudo patch -p3 < ~/Downloads/626.patch`

**Comment by Ploddles** [comment]
> I put in the colon. Perhaps I'm missing something else...
> 
> Update:
> It appears that the line the won't let it start up is:
> if scheme == 'https':
> 
> When I put that in, it no longer runs.
> Not sure what is happening. Editing with NotePad++

Did you manage to solve this?

I'm running it on a Mac but get the same result as you, ie the edit in util.py just causes gpodder to open and immediately close again.

**Comment by auouymous** [comment]
Python is whitespace sensitive, are you preserving the correct number of indentation spaces from the patch? Using tabs instead of spaces will also fail.

I don't know if python cares, but the files use LF line terminators and windows and mac like to use CRLF and CR. Setting your editor to output in unix LF may or may not fix the issue.

**Comment by Ploddles** [comment]
Thanks for your response.
I've had a little play.

Removing blank lines I had put in so I could find the edits quickly didn't help on it's own but what eventually solved it was removing the colons from the end of the 2 lines

1) if scheme == 'https':
and
2) else:

then retyping them.

I had originally copy and pasted the edits from the first post so I'm not sure how this fixed it but at least it is now fixed and up and running.

Thanks again for the original fix and responding to my query.

**Comment by lorditchy** [comment]
the problem for me was that the "conn" lines need the indentation like in auouymous post (1st post) so if you copy paste from the Jennifer202 post you might lose that indentation.

**Comment by elelay** [comment]
This is fixed in gPodder 3.10.9.
Thanks for the great community self-support on this issue!
